diff -Nru bladeenc-0.94.2/bladeenc/bladesys.c bladeenc-0.94.2.new/bladeenc/bladesys.c
--- bladeenc-0.94.2/bladeenc/bladesys.c	Tue May  8 12:11:53 2001
+++ bladeenc-0.94.2.new/bladeenc/bladesys.c	Sun Mar 23 18:05:30 2003
@@ -680,15 +680,29 @@
 
 
 #ifdef	UNIX_SYSTEM
-	pTmp = getenv ("HOME");
+	pTmp = getenv ("CONFIG_DIR");
 	if (pTmp != NULL)
 	{
 		strcpy (wpPath, pTmp);
 		if (wpPath[strlen(wpPath)-1] != '/')
 			strcat (wpPath, "/");
 
-		strcat (wpPath, ".bladeencrc");
+		strcat (wpPath, "bladeencrc");
+	}
+	else
+	{
+		pTmp = getenv ("HOME");
+		if (pTmp != NULL)
+		{
+			strcpy (wpPath, pTmp);
+			if (wpPath[strlen(wpPath)-1] != '/')
+				strcat (wpPath, "/");
 
+			strcat (wpPath, ".bladeencrc");
+		}
+	}
+	if (pTmp != NULL)
+	{	
 		fp = fopen (wpPath, "r");
 		if (fp != NULL)
 		{
